### PR TITLE
Remove Loader repository dependency from VVL

### PR DESCRIPTION
**Pull request by mark-lunarg**
ValidationLayers repo has no direct dependency upon the Vulkan-Loader repo.

**Note that this cannot be pushed WITHOUT supporting changes in Google and LunarG internal CI.**

### CODE REVIEW

**Comment by KarenGhavam-lunarG** [comment]
@mark-lunarg so once this goes through, the tests will expect a loader installed on the system to be able to execute (vs. using the loader built by this repository), true? Are there situations to correctly test the validation layers that would require you to have a newer loader than what might be installed on the system?

**Comment by lenny-lunarg** [comment]
This isn't how I thought we were going to remove the loader dependency long-term, based on earlier discussions. This change looks like me like all that's happening is that the loader repo build is being moved outside of update_deps. But that doesn't lift the dependency that the tests have on the loader.

I had been under the impression that we would stop using the entry points that are exported by the loader. Instead we would build a dispatch table (I imagine that the dispatch tables that are currently used in the layers would be just fine). Then we call into the loader using those dispatch tables. This means that for instance creation, for example, the current method is:

```cpp
VkResult result = vkCreateInstance(&create_info, nullptr, &instance);
```

while the new, improved method would be:

```cpp
VkResult result = dispatch_table->CreateInstance(&create_info, nullptr, &instance);
```

Note that in this example, the above code is intended to be function pointer in a struct &mdash; not a member function.

I think that the solution I described would do a far better job of removing the loader dependency, though I admit it would require a large number of changes to the tests, even though those changes would be quite simple.

**Comment by mark-lunarg** [comment]
Yep, you're right.  Had an overlooked loader install that made it seem like things were happy.

**Comment by lenny-lunarg** [comment]
**Note: I wrote this whole reply and then @mark-lunarg pre-empted this right before I hit send. I'm still going to post it in case the details are still relevant, but it sounds like Mark is already on it.**

The more I look at this the more I realize there's something I missed before writing my comment above, because I didn't understand how this works at all on Linux. It does (CI demonstrates that), but I still feel this isn't the change I would have expected.

In particular, the CMake `add_library` command is being removed from this repo. But I think you're still finding the loader through the `find_package` command [here](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/markl-loaderless-vvl/tests/CMakeLists.txt#L85). Even with your changes, `FindVulkan.cmake` still searches for the loader [here](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/markl-loaderless-vvl/cmake/FindVulkan.cmake#L54).

So I think this is still finding the loader at buid time &mdash; it's just finding it through the `-DVULKAN_LOADER_INSTALL_DIR` and `find_package(vulkan)` combination instead of through update_deps. I still think the solution is what I described in my previous post.

**Comment by mark-lunarg** [comment]
Changed the LVTs to load core vulkan entrypoints dynamically. This should address @lenny-lunarg's concerns

**Comment by mark-lunarg** [comment]
Pushed additional commit to fix clang-format bug which caused travis failure.

File: tests/vklayertests_command.cpp [hunk 1]
```diff
@@ -356,7 +356,7 @@ TEST_F(VkLayerTest, IndexBufferBadBindOffset) {
 }
 
 TEST_F(VkLayerTest, MissingClearAttachment) {
-    TEST_DESCRIPTION("Points to a wrong colorAttachment index in a VkClearAttachment structure passed to vkCmdClearAttachments");
+    TEST_DESCRIPTION("Points to a wrong colorAttachment index in a VkClearAttachment structure passed to vk::CmdClearAttachments");
```
Comment (Tony-LunarG) [inline tests/vklayertests_command.cpp]: No vk::

File: tests/vklayertests_command.cpp [hunk 2]
```diff
@@ -764,13 +767,13 @@ TEST_F(VkLayerTest, ClearColorAttachmentsZeroLayercount) {
...
     VkClearRect clear_rect = {{{0, 0}, {32, 32}}};
-    vkCmdClearAttachments(m_commandBuffer->handle(), 1, &color_attachment, 1, &clear_rect);
+    vk::CmdClearAttachments(m_commandBuffer->handle(), 1, &color_attachment, 1, &clear_rect);
 
     m_errorMonitor->VerifyFound();
 }
 
 TEST_F(VkLayerTest, ExecuteCommandsPrimaryCB) {
-    TEST_DESCRIPTION("Attempt vkCmdExecuteCommands with a primary command buffer (should only be secondary)");
+    TEST_DESCRIPTION("Attempt vk::CmdExecuteCommands with a primary command buffer (should only be secondary)");
```
Comment (Tony-LunarG) [inline tests/vklayertests_command.cpp]: No vk::

File: tests/vklayertests_command.cpp [hunk 3]
```diff
@@ -3780,20 +3807,20 @@ TEST_F(VkLayerTest, CommandQueueFlags) {
 }
 
 TEST_F(VkLayerTest, ExecuteUnrecordedSecondaryCB) {
-    TEST_DESCRIPTION("Attempt vkCmdExecuteCommands with a CB in the initial state");
+    TEST_DESCRIPTION("Attempt vk::CmdExecuteCommands with a CB in the initial state");
```
Comment (Tony-LunarG) [inline tests/vklayertests_command.cpp]: No vk:: here and ExecuteSecondaryCBWithLayout mismatch

File: tests/vklayertests_command.cpp [hunk 4]
```diff
@@ -3857,13 +3884,13 @@ TEST_F(VkLayerTest, ExecuteSecondaryCBWithLayoutMismatch) {
...
     m_errorMonitor->ExpectSuccess();
-    vkCmdExecuteCommands(m_commandBuffer->handle(), 1, &secondary.handle());
+    vk::CmdExecuteCommands(m_commandBuffer->handle(), 1, &secondary.handle());
     m_errorMonitor->VerifyNotFound();
     m_commandBuffer->end();
 }
 
 TEST_F(VkLayerTest, SetDynViewportParamTests) {
-    TEST_DESCRIPTION("Test parameters of vkCmdSetViewport without multiViewport feature");
+    TEST_DESCRIPTION("Test parameters of vk::CmdSetViewport without multiViewport feature");
```
Comment (Tony-LunarG) [inline tests/vklayertests_command.cpp]: No vk::

File: tests/vklayertests_command.cpp [hunk 5]
```diff
@@ -3966,7 +3993,7 @@ TEST_F(VkLayerTest, SetDynViewportParamMaintenance1Tests) {
 }
 
 TEST_F(VkLayerTest, SetDynViewportParamMultiviewportTests) {
-    TEST_DESCRIPTION("Test parameters of vkCmdSetViewport with multiViewport feature enabled");
+    TEST_DESCRIPTION("Test parameters of vk::CmdSetViewport with multiViewport feature enabled");
```
Comment (Tony-LunarG) [inline tests/vklayertests_command.cpp]: No  vk::

File: tests/vklayertests_command.cpp [hunk 6]
```diff
@@ -4242,7 +4269,7 @@ TEST_F(VkLayerTest, PushDescriptorSetCmdPushBadArgs) {
 }
 
 TEST_F(VkLayerTest, SetDynScissorParamTests) {
-    TEST_DESCRIPTION("Test parameters of vkCmdSetScissor without multiViewport feature");
+    TEST_DESCRIPTION("Test parameters of vk::CmdSetScissor without multiViewport feature");
```
Comment (Tony-LunarG) [inline tests/vklayertests_command.cpp]: No vk::

File: tests/vklayertests_command.cpp [hunk 7]
```diff
@@ -4290,15 +4317,15 @@ TEST_F(VkLayerTest, SetDynScissorParamTests) {
...
+        vk::CmdSetScissor(m_commandBuffer->handle(), 0, 1, &test_case.scissor);
         m_errorMonitor->VerifyFound();
     }
 
     m_commandBuffer->end();
 }
 
 TEST_F(VkLayerTest, SetDynScissorParamMultiviewportTests) {
-    TEST_DESCRIPTION("Test parameters of vkCmdSetScissor with multiViewport feature enabled");
+    TEST_DESCRIPTION("Test parameters of vk::CmdSetScissor with multiViewport feature enabled");
```
Comment (Tony-LunarG) [inline tests/vklayertests_command.cpp]: No vk::

File: tests/vklayertests_command.cpp [hunk 8]
```diff
@@ -4328,25 +4355,25 @@ TEST_F(VkLayerTest, SetDynScissorParamMultiviewportTests) {
...
         m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, "VUID-vkCmdSetScissor-scissorCount-arraylength");
-        vkCmdSetScissor(m_commandBuffer->handle(), 1, 0, scissors.data());
+        vk::CmdSetScissor(m_commandBuffer->handle(), 1, 0, scissors.data());
         m_errorMonitor->VerifyFound();
     }
 }
 
 TEST_F(VkLayerTest, DrawIndirect) {
-    TEST_DESCRIPTION("Test covered valid usage for vkCmdDrawIndirect");
+    TEST_DESCRIPTION("Test covered valid usage for vk::CmdDrawIndirect");
```
Comment (Tony-LunarG) [inline tests/vklayertests_command.cpp]: No vk::

Comment (mark-lunarg) [reply tests/vklayertests_command.cpp]: Fixed.

**Comment by mark-lunarg** [comment]
Rebased, CI vetted and fixed, pushed to Master.

**Pull request closed without merging**